components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserRealmProxy.java

```diff
@@ -665,6 +665,17 @@ private UserStoreInfo getUserStoreInfo(RealmConfiguration realmConfig,
             info.setBulkImportSupported(this.isBulkImportSupported());
             info.setDomainName(realmConfig.getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_DOMAIN_NAME));
 
+            boolean caseSensitiveUsername;
+            String caseInsensitiveUsernameString;
+            caseInsensitiveUsernameString = manager.getRealmConfiguration().getUserStoreProperty
+                    ("CaseInsensitiveUsername");
+
+            if (caseInsensitiveUsernameString != null) {
+                caseSensitiveUsername = !Boolean.parseBoolean(caseInsensitiveUsernameString);
+            } else {
+                caseSensitiveUsername = true;
```

Comment:
Boolean.parseBoolean() returns false for null values, so this null check is redundant. 673-676 could have simplified as

caseSensitiveUsername = !Boolean.parseBoolean(caseInsensitiveUsernameString);